File: frontend/src/constants/bookTypes.ts
/**
 * Book Types Configuration
 * Matches backend book_types.py
 */

export interface BookType {
  id: string;
  name: string;
  category: 'technical' | 'nonfiction' | 'creative' | 'reference';
  description: string;
  icon: string; // Emoji or icon name
  availableTones: string[];
  typicalChapterRange: [number, number];
  recommendedFeatures: string[];
}

export const BOOK_TYPES: BookType[] = [
  // TECHNICAL BOOKS
  {
    id: 'programming_tutorial',
    name: 'Programming Tutorial',
    category: 'technical',
    description: 'Step-by-step coding guide with executable examples',
    icon: '💻',
    availableTones: ['conversational', 'professional', 'academic'],
    typicalChapterRange: [8, 15],
    recommendedFeatures: [
      'code_blocks',
      'interactive_plots',
      'quizzes',
      'tabs',
      'dropdowns',
      'admonitions',
      'cross_references'
    ]
  },
  {
    id: 'data_science',
    name: 'Data Science & Analytics',
    category: 'technical',
    description: 'Statistical analysis and ML guides with visualizations',
    icon: '📊',
    availableTones: ['academic', 'professional'],
    typicalChapterRange: [10, 20],
    recommendedFeatures: [
      'code_blocks',
      'math_equations',
      'interactive_plots',
      'tables',
      'theorems',
      'admonitions',
      'figures'
    ]
  },
  {
    id: 'academic_course',
    name: 'Academic Course',
    category: 'technical',
    description: 'Comprehensive educational course material',
    icon: '🎓',
    availableTones: ['academic', 'professional'],
    typicalChapterRange: [12, 25],
    recommendedFeatures: [
      'code_blocks',
      'math_equations',
      'theorems',
      'quizzes',
      'citations',
      'cross_references',
      'admonitions'
    ]
  },
  {
    id: 'research_paper',
    name: 'Research Paper',
    category: 'technical',
    description: 'Academic research with citations and proofs',
    icon: '🔬',
    availableTones: ['academic'],
    typicalChapterRange: [5, 10],
    recommendedFeatures: [
      'math_equations',
      'theorems',
      'citations',
      'figures',
      'tables',
      'cross_references'
    ]
  },
  {
    id: 'technical_documentation',
    name: 'Technical Documentation',
    category: 'technical',
    description: 'API docs, developer guides, and technical references',
    icon: '📖',
    availableTones: ['professional', 'technical'],
    typicalChapterRange: [5, 15],
    recommendedFeatures: [
      'code_blocks',
      'tabs',
      'admonitions',
      'tables',
      'cross_references',
      'dropdowns'
    ]
  },

  // NON-FICTION BOOKS
  {
    id: 'business_book',
    name: 'Business Book',
    category: 'nonfiction',
    description: 'Business strategies, leadership, and management',
    icon: '💼',
    availableTones: ['professional', 'conversational', 'inspirational'],
    typicalChapterRange: [10, 20],
    recommendedFeatures: [
      'admonitions',
      'cards',
      'dropdowns',
      'figures',
      'quizzes'
    ]
  },
  {
    id: 'how_to_guide',
    name: 'How-To Guide',
    category: 'nonfiction',
    description: 'Practical guides and skill-building handbooks',
    icon: '🛠️',
    availableTones: ['conversational', 'professional'],
    typicalChapterRange: [8, 15],
    recommendedFeatures: [
      'admonitions',
      'dropdowns',
      'figures',
      'quizzes',
      'task_lists'
    ]
  },
  {
    id: 'history_biography',
    name: 'History & Biography',
    category: 'nonfiction',
    description: 'Historical accounts and biographical narratives',
    icon: '📜',
    availableTones: ['narrative', 'academic', 'journalistic'],
    typicalChapterRange: [10, 25],
    recommendedFeatures: [
      'figures',
      'citations',
      'cross_references',
      'dropdowns'
    ]
  },
  {
    id: 'science_medicine',
    name: 'Science & Medicine',
    category: 'nonfiction',
    description: 'Scientific concepts and medical topics',
    icon: '⚕️',
    availableTones: ['academic', 'professional', 'accessible'],
    typicalChapterRange: [8, 20],
    recommendedFeatures: [
      'figures',
      'math_equations',
      'citations',
      'admonitions',
      'tables'
    ]
  },
  {
    id: 'reference_guide',
    name: 'Reference Guide',
    category: 'reference',
    description: 'Comprehensive reference material and lookup guides',
    icon: '📚',
    availableTones: ['professional', 'technical'],
    typicalChapterRange: [5, 30],
    recommendedFeatures: [
      'tables',
      'cross_references',
      'dropdowns',
      'tabs',
      'admonitions'
    ]
  },

  // CREATIVE
  {
    id: 'fiction_novel',
    name: 'Fiction Novel',
    category: 'creative',
    description: 'Creative fiction and storytelling',
    icon: '📕',
    availableTones: ['narrative', 'literary', 'conversational'],
    typicalChapterRange: [10, 30],
    recommendedFeatures: [
      'dropdowns',
      'admonitions'
    ]
  },
  {
    id: 'essay_collection',
    name: 'Essay Collection',
    category: 'creative',
    description: 'Collections of essays and thought pieces',
    icon: '✍️',
    availableTones: ['literary', 'conversational', 'academic'],
    typicalChapterRange: [5, 20],
    recommendedFeatures: [
      'citations',
      'cross_references',
      'admonitions'
    ]
  },

  // REFERENCE/SPECIALIZED
  {
    id: 'childrens_educational',
    name: "Children's Educational",
    category: 'reference',
    description: 'Educational content for children',
    icon: '🎨',
    availableTones: ['playful', 'simple', 'encouraging'],
    typicalChapterRange: [5, 15],
    recommendedFeatures: [
      'figures',
      'admonitions',
      'quizzes',
      'cards',
      'task_lists'
    ]
  },
  {
    id: 'white_paper',
    name: 'White Paper',
    category: 'reference',
    description: 'Authoritative reports and technical briefs',
    icon: '📄',
    availableTones: ['professional', 'technical', 'academic'],
    typicalChapterRange: [3, 8],
    recommendedFeatures: [
      'figures',
      'tables',
      'citations',
      'math_equations'
    ]
  },
  {
    id: 'cookbook_patterns',
    name: 'Cookbook/Patterns',
    category: 'reference',
    description: 'Recipe-style solutions and design patterns',
    icon: '🍳',
    availableTones: ['professional', 'conversational'],
    typicalChapterRange: [10, 30],
    recommendedFeatures: [
      'code_blocks',
      'tabs',
      'admonitions',
      'cross_references',
      'dropdowns'
    ]
  }
];

export const TONE_OPTIONS = [
  { value: 'conversational', label: 'Conversational', description: 'Friendly and approachable' },
  { value: 'professional', label: 'Professional', description: 'Clear and business-like' },
  { value: 'academic', label: 'Academic', description: 'Scholarly and formal' },
  { value: 'technical', label: 'Technical', description: 'Precise and detailed' },
  { value: 'narrative', label: 'Narrative', description: 'Story-driven' },
  { value: 'inspirational', label: 'Inspirational', description: 'Motivating and uplifting' },
  { value: 'playful', label: 'Playful', description: 'Fun and engaging' },
  { value: 'simple', label: 'Simple', description: 'Easy to understand' },
  { value: 'literary', label: 'Literary', description: 'Artistic and expressive' },
  { value: 'journalistic', label: 'Journalistic', description: 'Factual and objective' },
  { value: 'accessible', label: 'Accessible', description: 'Easy for general audiences' },
  { value: 'encouraging', label: 'Encouraging', description: 'Supportive and positive' }
];

export const AUDIENCE_OPTIONS = [
  { value: 'beginner', label: 'Beginner', description: 'New to the topic' },
  { value: 'intermediate', label: 'Intermediate', description: 'Some experience' },
  { value: 'advanced', label: 'Advanced', description: 'Expert level' },
  { value: 'general', label: 'General Public', description: 'No prior knowledge assumed' },
  { value: 'professional', label: 'Professionals', description: 'Working in the field' },
  { value: 'academic', label: 'Academics/Students', description: 'Educational setting' },
  { value: 'children', label: 'Children', description: 'Age 8-12' },
  { value: 'teens', label: 'Teens', description: 'Age 13-17' }
];

export function getBookType(id: string): BookType | undefined {
  return BOOK_TYPES.find(type => type.id === id);
}

export function getBookTypesByCategory(category: string): BookType[] {
  return BOOK_TYPES.filter(type => type.category === category);
}

export function getToneOptionsForBookType(bookTypeId: string): Array<{value: string, label: string, description: string, recommended?: boolean}> {
  const bookType = getBookType(bookTypeId);
  if (!bookType) return TONE_OPTIONS;

  // Return ALL tones, but mark the recommended ones
  return TONE_OPTIONS.map(tone => ({
    ...tone,
    recommended: bookType.availableTones.includes(tone.value)
  }));
}

File: backend/book_types.py
"""
Book Types Configuration for LiquidBooks

Defines all supported book types with their system prompts, metadata, and requirements.
"""

from typing import Dict, List, Optional
from dataclasses import dataclass


@dataclass
class BookType:
    """Configuration for a specific book type"""
    id: str
    name: str
    category: str  # 'technical', 'nonfiction', 'creative', 'reference'
    description: str
    system_prompt: str
    available_tones: List[str]
    typical_chapter_range: tuple[int, int]
    recommended_features: List[str]


# System prompts for each book type
BOOK_TYPE_PROMPTS = {
    'programming_tutorial': """You are an expert technical writer creating a programming tutorial book.

WRITING STYLE:
- Conversational and encouraging tone
- Explain concepts clearly for beginners
- Use real-world examples, not foo/bar
- Include practical, runnable code
- Explain WHY, not just HOW

STRUCTURE REQUIREMENTS:
- Start with prerequisites and setup
- Build concepts progressively
- Each chapter builds on previous knowledge
- Include hands-on exercises
- Add solutions in expandable sections
- End chapters with quizzes

CODE REQUIREMENTS:
- All code must be complete and executable
- Use proper syntax highlighting
- Include explanatory comments
- Show expected output
- Discuss common pitfalls

JUPYTER BOOK FEATURES TO USE:
- Code blocks with language tags
- Admonitions for tips, warnings, notes
- Dropdowns for exercise solutions
- Quizzes at chapter ends
- Cross-references between chapters

Make it engaging, practical, and confidence-building for learners.""",

    'data_science': """You are an expert data scientist and educator creating a data science book.

WRITING STYLE:
- Balance theory and practical application
- Use precise technical language
- Explain statistical concepts clearly
- Include mathematical foundations

STRUCTURE REQUIREMENTS:
- Start with problem/question
- Show data exploration and visualization
- Explain methodology and approach
- Implement solution with code
- Interpret results and implications
- Discuss limitations and assumptions

CONTENT REQUIREMENTS:
- Real datasets (or realistic synthetic data)
- Complete data pipeline (load, clean, analyze, visualize)
- Mathematical equations where relevant
- Statistical significance discussions
- Reproducible analysis

JUPYTER BOOK FEATURES TO USE:
- Code blocks for data analysis
- Math equations (LaTeX)
- Interactive plots (describe Plotly/Altair code)
- Tables for data summaries
- Admonitions for important notes
- Theorem blocks for statistical concepts

Make it rigorous yet accessible.""",

    'academic_course': """You are an experienced educator creating structured course material.

WRITING STYLE:
- Clear, academic tone
- Well-organized and logical flow
- Progressive difficulty
- Explicit learning objectives

STRUCTURE REQUIREMENTS:
- Module overview with learning objectives
- Lecture content with clear explanations
- Worked examples showing process
- Practice exercises for students
- Formative assessments (quizzes)
- Connect to previous and next modules
- Summaries and key takeaways

PEDAGOGICAL APPROACH:
- State learning objectives upfront
- Build on prerequisite knowledge
- Include multiple representations (text, diagrams, code)
- Provide immediate feedback (quizzes)
- Encourage active learning

JUPYTER BOOK FEATURES TO USE:
- Code blocks for demonstrations
- Math equations for formulas
- Admonitions for learning objectives and key points
- Quizzes for assessment
- Cross-references to related concepts
- Launch buttons (Binder/Colab) for hands-on practice

Make it pedagogically sound and student-centered.""",

    'research_paper': """You are writing an academic research paper following scholarly standards.

WRITING STYLE:
- Formal academic tone
- Objective and evidence-based
- Precise technical language
- Proper citations throughout

REQUIRED STRUCTURE:
1. Abstract (150-250 words)
   - Problem statement
   - Method overview
   - Key results
   - Conclusion

2. Introduction
   - Background and context
   - Motivation for research
   - Research questions
   - Contributions

3. Related Work
   - Survey of existing approaches
   - Compare and contrast methods
   - Identify gaps

4. Methodology
   - Detailed approach description
   - Mathematical formulations
   - Algorithm descriptions
   - Experimental setup

5. Results
   - Present findings objectively
   - Use tables and figures
   - Statistical analysis

6. Discussion
   - Interpret results
   - Compare to prior work
   - Discuss implications
   - Acknowledge limitations

7. Conclusion
   - Summarize contributions
   - Future work directions

8. References
   - Proper academic citations

JUPYTER BOOK FEATURES TO USE:
- Math equations with labels
- Theorem/Proof blocks
- Figures with captions and references
- Tables for results
- Citations and bibliography
- Algorithm blocks
- Cross-references

Make it rigorous and scholarly.""",

    'technical_documentation': """You are creating professional technical documentation.

WRITING STYLE:
- Clear and concise
- Task-oriented
- Professional tone
- Scannable structure

STRUCTURE REQUIREMENTS:
- Getting Started guide
- Installation instructions
- Architecture overview
- Feature documentation
- API reference
- Code examples
- Troubleshooting guide
- FAQ

CONTENT REQUIREMENTS:
- Step-by-step instructions
- Clear prerequisites
- Complete code examples
- Common use cases
- Error messages and solutions
- Best practices

JUPYTER BOOK FEATURES TO USE:
- Code blocks with copy buttons
- Admonitions for warnings and tips
- Dropdowns for advanced topics
- Tabs for language variations
- Definition lists for terminology
- Cross-references between sections

Make it practical and easy to navigate.""",

    'business_book': """You are writing an engaging business and professional development book.

WRITING STYLE:
- Conversational and accessible
- Inspirational yet practical
- Use storytelling
- Relatable examples

STRUCTURE REQUIREMENTS:
- Hook with relatable problem
- Present framework or principles
- One principle per chapter
- Real-world case studies
- Actionable strategies
- Reflection questions
- Chapter summaries

CONTENT APPROACH:
- Start with reader's pain points
- Provide clear frameworks
- Support with evidence and stories
- Include expert insights
- Practical action steps
- Avoid jargon, explain concepts simply

JUPYTER BOOK FEATURES TO USE:
- Cards for key insights
- Blockquotes for important points
- Admonitions for tips and warnings
- Dropdowns for exercises/worksheets
- Tables for comparisons
- Checklists for action items

Make it engaging, practical, and transformative.""",

    'fiction_novel': """You are writing a fiction novel with strong narrative and characters.

WRITING STYLE:
- Narrative voice consistent with tone
- Show, don't tell
- Vivid descriptions
- Natural dialogue
- Emotional engagement

CRAFT REQUIREMENTS:
- Strong character development
- Compelling plot progression
- Scene-by-scene structure
- Proper pacing
- Satisfying chapter endings
- Thematic consistency

STORY ELEMENTS:
- Well-developed characters with clear motivations
- Conflict and tension
- Setting descriptions
- Dialogue that reveals character
- Sensory details
- Emotional beats

JUPYTER BOOK FEATURES TO USE:
- Epigraphs for chapter quotes
- Margin notes for author commentary (if desired)
- Dropdowns for author's notes
- Cross-references for character/location tracking

Make it immersive and emotionally resonant.""",

    'reference_guide': """You are creating a comprehensive reference guide.

WRITING STYLE:
- Concise and precise
- Consistent format
- Easy to scan
- Authoritative tone

STRUCTURE REQUIREMENTS:
- Logical organization (alphabetical or topical)
- Clear entry structure
- Cross-references
- Complete coverage
- Index-friendly

ENTRY FORMAT:
- Term or concept name
- Clear definition
- Syntax or usage (if applicable)
- Parameters or options
- Examples
- Related concepts
- Notes or warnings

JUPYTER BOOK FEATURES TO USE:
- Definition lists
- Target headers for labels
- Cross-references extensively
- Code blocks for syntax
- Tables for parameter lists
- Admonitions for warnings
- Glossary terms

Make it comprehensive and easy to search.""",

    'how_to_guide': """You are writing a practical how-to guide.

WRITING STYLE:
- Direct and instructional
- Friendly and encouraging
- Step-by-step clarity
- Beginner-friendly

STRUCTURE REQUIREMENTS:
- Clear goal statement
- Prerequisites and materials
- Step-by-step instructions
- Troubleshooting section
- Tips and tricks
- Variations or alternatives

INSTRUCTION FORMAT:
- Numbered steps
- One action per step
- Include expected results
- Highlight critical points
- Provide visual descriptions
- Common mistakes to avoid

JUPYTER BOOK FEATURES TO USE:
- Numbered lists for steps
- Admonitions for warnings and tips
- Images for visual reference
- Dropdowns for advanced variations
- Task lists for checklists
- Tables for materials/tools

Make it clear, actionable, and confidence-building.""",

    'history_biography': """You are writing historical or biographical content.

WRITING STYLE:
- Narrative and engaging
- Well-researched and accurate
- Contextual background
- Chronological or thematic

STRUCTURE REQUIREMENTS:
- Set historical context
- Chronological progression or thematic chapters
- Key events and their significance
- Analysis and interpretation
- Primary source references
- Timeline summary

CONTENT APPROACH:
- Balance storytelling with facts
- Provide context and background
- Explain significance and impact
- Include diverse perspectives
- Use vivid descriptions
- Connect to larger themes

JUPYTER BOOK FEATURES TO USE:
- Blockquotes for primary sources
- Images and figures
- Timeline visualizations
- Definition lists for key terms
- Citations and references
- Cross-references between events

Make it informative and engaging.""",

    'science_medicine': """You are writing about scientific or medical topics.

WRITING STYLE:
- Clear and accurate
- Evidence-based
- Appropriate technical level
- Accessible explanations

STRUCTURE REQUIREMENTS:
- Introduction to topic and significance
- Scientific foundation and principles
- Current research and findings
- Practical applications
- Case studies or examples
- Future directions

CONTENT APPROACH:
- Define technical terms
- Use analogies for complex concepts
- Cite research studies
- Explain methodology
- Discuss implications
- Address common misconceptions

JUPYTER BOOK FEATURES TO USE:
- Math equations for formulas
- Figures with detailed captions
- Tables for data
- Admonitions for important notes
- Citations and bibliography
- Definition lists for terminology
- Interactive plots for data visualization

Make it accurate, clear, and trustworthy.""",

    'essay_collection': """You are writing a collection of essays.

WRITING STYLE:
- Thoughtful and analytical
- Clear argumentation
- Personal voice (if appropriate)
- Engaging prose

STRUCTURE REQUIREMENTS:
- Introduction with thesis
- Well-organized arguments
- Evidence and examples
- Counter-arguments addressed
- Conclusion with implications
- Connections between essays

CONTENT APPROACH:
- Clear thesis statement
- Logical argument structure
- Support with evidence
- Cite sources appropriately
- Reflect on implications
- Synthesize ideas across essays

JUPYTER BOOK FEATURES TO USE:
- Blockquotes for key ideas
- Footnotes for additional thoughts
- Citations and references
- Cross-references between essays
- Admonitions for key arguments

Make it thought-provoking and well-argued.""",

    'childrens_educational': """You are writing educational content for children.

WRITING STYLE:
- Simple, clear language
- Age-appropriate vocabulary
- Encouraging and fun
- Interactive and engaging

STRUCTURE REQUIREMENTS:
- Start with hook or question
- Short, focused sections
- Visual descriptions
- Hands-on activities
- Review and reinforcement
- Fun facts and surprises

CONTENT APPROACH:
- Explain concepts simply
- Use relatable examples
- Include activities and games
- Ask engaging questions
- Celebrate learning
- Build confidence

JUPYTER BOOK FEATURES TO USE:
- Images and figures
- Cards for visual appeal
- Admonitions for fun facts
- Quizzes with encouraging feedback
- Interactive widgets (if applicable)
- Simple code examples (for coding books)

Make it fun, engaging, and age-appropriate.""",

    'white_paper': """You are writing a professional white paper.

WRITING STYLE:
- Professional and authoritative
- Data-driven
- Objective analysis
- Clear recommendations

STRUCTURE REQUIREMENTS:
1. Executive Summary
2. Problem Statement
3. Background and Context
4. Current Situation Analysis
5. Proposed Solution
6. Implementation Approach
7. Benefits and ROI
8. Conclusion and Recommendations

CONTENT APPROACH:
- Lead with key findings
- Support with data and research
- Present objective analysis
- Provide actionable recommendations
- Use visuals for data
- Include case studies
- Address objections

JUPYTER BOOK FEATURES TO USE:
- Figures and charts
- Tables for data
- Admonitions for key findings
- Citations and references
- Cross-references
- Definition lists for terminology

Make it professional, credible, and actionable.""",

    'cookbook_patterns': """You are writing a cookbook or pattern collection.

WRITING STYLE:
- Clear and practical
- Consistent format
- Problem-solution oriented
- Efficient and reusable

STRUCTURE REQUIREMENTS:
- Organized by category
- One recipe/pattern per page
- Clear problem statement
- Complete solution
- Variations and alternatives
- Usage examples

RECIPE/PATTERN FORMAT:
- Name and description
- Problem it solves
- Prerequisites or ingredients
- Step-by-step solution
- Complete code example
- Discussion and notes
- Related patterns

JUPYTER BOOK FEATURES TO USE:
- Code blocks with explanations
- Admonitions for tips and warnings
- Tabs for variations
- Cross-references to related patterns
- Definition lists for parameters
- Tables for comparisons

Make it practical and reusable.""",
}


# Book type metadata
BOOK_TYPES: Dict[str, BookType] = {
    'programming_tutorial': BookType(
        id='programming_tutorial',
        name='Programming Tutorial',
        category='technical',
        description='Step-by-step coding guide with executable examples',
        system_prompt=BOOK_TYPE_PROMPTS['programming_tutorial'],
        available_tones=['conversational', 'professional', 'academic'],
        typical_chapter_range=(8, 15),
        recommended_features=[
            'code_blocks', 'interactive_plots', 'quizzes', 'tabs',
            'dropdowns', 'admonitions', 'cross_references'
        ]
    ),
    'data_science': BookType(
        id='data_science',
        name='Data Science & Analytics',
        category='technical',
        description='Statistical analysis and ML guides with visualizations',
        system_prompt=BOOK_TYPE_PROMPTS['data_science'],
        available_tones=['academic', 'professional'],
        typical_chapter_range=(10, 20),
        recommended_features=[
            'code_blocks', 'math_equations', 'interactive_plots', 'tables',
            'theorems', 'admonitions', 'figures'
        ]
    ),
    'academic_course': BookType(
        id='academic_course',
        name='Academic Course',
        category='technical',
        description='Structured course with modules and assessments',
        system_prompt=BOOK_TYPE_PROMPTS['academic_course'],
        available_tones=['academic', 'professional'],
        typical_chapter_range=(12, 24),
        recommended_features=[
            'code_blocks', 'math_equations', 'quizzes', 'admonitions',
            'cross_references', 'binder', 'colab', 'dropdowns'
        ]
    ),
    'research_paper': BookType(
        id='research_paper',
        name='Research Paper',
        category='technical',
        description='Academic research with methodology and findings',
        system_prompt=BOOK_TYPE_PROMPTS['research_paper'],
        available_tones=['academic'],
        typical_chapter_range=(6, 10),
        recommended_features=[
            'math_equations', 'theorems', 'proofs', 'citations',
            'figures', 'tables', 'cross_references', 'algorithms'
        ]
    ),
    'technical_documentation': BookType(
        id='technical_documentation',
        name='Technical Documentation',
        category='technical',
        description='Software documentation and API guides',
        system_prompt=BOOK_TYPE_PROMPTS['technical_documentation'],
        available_tones=['professional', 'conversational'],
        typical_chapter_range=(10, 30),
        recommended_features=[
            'code_blocks', 'admonitions', 'dropdowns', 'tabs',
            'definition_lists', 'cross_references', 'tables'
        ]
    ),
    'business_book': BookType(
        id='business_book',
        name='Business & Self-Help',
        category='nonfiction',
        description='Professional development and business strategies',
        system_prompt=BOOK_TYPE_PROMPTS['business_book'],
        available_tones=['conversational', 'professional', 'inspirational'],
        typical_chapter_range=(10, 15),
        recommended_features=[
            'cards', 'admonitions', 'blockquotes', 'dropdowns', 'tables'
        ]
    ),
    'fiction_novel': BookType(
        id='fiction_novel',
        name='Fiction Novel',
        category='creative',
        description='Narrative fiction with plot and characters',
        system_prompt=BOOK_TYPE_PROMPTS['fiction_novel'],
        available_tones=['dramatic', 'humorous', 'dark', 'lighthearted'],
        typical_chapter_range=(15, 40),
        recommended_features=[
            'epigraphs', 'dropdowns', 'margin_notes', 'cross_references'
        ]
    ),
    'reference_guide': BookType(
        id='reference_guide',
        name='Reference Guide',
        category='reference',
        description='Comprehensive reference material',
        system_prompt=BOOK_TYPE_PROMPTS['reference_guide'],
        available_tones=['professional', 'academic'],
        typical_chapter_range=(20, 50),
        recommended_features=[
            'definition_lists', 'cross_references', 'glossary',
            'target_headers', 'tables', 'code_blocks'
        ]
    ),
    'how_to_guide': BookType(
        id='how_to_guide',
        name='How-To & Practical Guide',
        category='nonfiction',
        description='Step-by-step instructions for skills/tasks',
        system_prompt=BOOK_TYPE_PROMPTS['how_to_guide'],
        available_tones=['conversational', 'professional'],
        typical_chapter_range=(8, 15),
        recommended_features=[
            'admonitions', 'dropdowns', 'images', 'lists', 'tables'
        ]
    ),
    'history_biography': BookType(
        id='history_biography',
        name='History & Biography',
        category='nonfiction',
        description='Historical accounts and life stories',
        system_prompt=BOOK_TYPE_PROMPTS['history_biography'],
        available_tones=['academic', 'conversational', 'narrative'],
        typical_chapter_range=(10, 20),
        recommended_features=[
            'images', 'figures', 'blockquotes', 'definition_lists',
            'citations', 'cross_references'
        ]
    ),
    'science_medicine': BookType(
        id='science_medicine',
        name='Science & Medicine',
        category='nonfiction',
        description='Scientific concepts and medical topics',
        system_prompt=BOOK_TYPE_PROMPTS['science_medicine'],
        available_tones=['academic', 'professional', 'conversational'],
        typical_chapter_range=(10, 18),
        recommended_features=[
            'math_equations', 'figures', 'tables', 'citations',
            'admonitions', 'interactive_plots', 'definition_lists'
        ]
    ),
    'essay_collection': BookType(
        id='essay_collection',
        name='Essay Collection',
        category='nonfiction',
        description='Compilation of essays on themes',
        system_prompt=BOOK_TYPE_PROMPTS['essay_collection'],
        available_tones=['academic', 'conversational', 'opinion'],
        typical_chapter_range=(8, 15),
        recommended_features=[
            'blockquotes', 'footnotes', 'citations', 'cross_references', 'admonitions'
        ]
    ),
    'childrens_educational': BookType(
        id='childrens_educational',
        name="Children's Educational",
        category='educational',
        description='Age-appropriate educational content for kids',
        system_prompt=BOOK_TYPE_PROMPTS['childrens_educational'],
        available_tones=['simple', 'fun', 'engaging'],
        typical_chapter_range=(5, 12),
        recommended_features=[
            'images', 'cards', 'admonitions', 'quizzes', 'widgets'
        ]
    ),
    'white_paper': BookType(
        id='white_paper',
        name='White Paper',
        category='nonfiction',
        description='In-depth report on topic or solution',
        system_prompt=BOOK_TYPE_PROMPTS['white_paper'],
        available_tones=['professional', 'academic'],
        typical_chapter_range=(6, 10),
        recommended_features=[
            'figures', 'tables', 'citations', 'cross_references',
            'admonitions', 'definition_lists'
        ]
    ),
    'cookbook_patterns': BookType(
        id='cookbook_patterns',
        name='Cookbook/Patterns',
        category='reference',
        description='Collection of recipes or code patterns',
        system_prompt=BOOK_TYPE_PROMPTS['cookbook_patterns'],
        available_tones=['conversational', 'professional'],
        typical_chapter_range=(15, 40),
        recommended_features=[
            'code_blocks', 'admonitions', 'tabs', 'cross_references',
            'definition_lists', 'tables'
        ]
    ),
}


def get_book_type(book_type_id: str) -> Optional[BookType]:
    """Get book type configuration by ID"""
    return BOOK_TYPES.get(book_type_id)


def get_all_book_types() -> List[BookType]:
    """Get all book type configurations"""
    return list(BOOK_TYPES.values())


def get_book_types_by_category(category: str) -> List[BookType]:
    """Get book types filtered by category"""
    return [bt for bt in BOOK_TYPES.values() if bt.category == category]
